crates/libs/core/src/handles.rs

```diff
@@ -0,0 +1,39 @@
+/// Custom code to free a handle.
+///
+/// This is similar to the `Drop` trait, and may be used to implement `Drop`, but allows handles
+/// to be dropped depending on context.
+pub trait Free {
+    /// Calls the handle's free function.
+    fn free(&mut self);
```

Comment:
This might need to be unsafe. An owned type asserts the right to be able to free. Without that, as you say, whether or not it's ok to free depends on the context. Which we can't, only the user can.

Reply:
Good call: https://github.com/microsoft/windows-rs/pull/3013/commits/5986214536be1b760f1378629c2ca7ecc51b869c